File: domain/club.go
package domain

type ClubMember struct {
	Icon struct {
		ID int `json:"id"`
	} `json:"icon"`
	Tag       string `json:"tag"`
	Name      string `json:"name"`
	Trophies  int    `json:"trophies"`
	Role      string `json:"role"`
	NameColor string `json:"nameColor"`
}

type Club struct {
	Tag              string       `json:"tag"`
	Name             string       `json:"name"`
	Description      string       `json:"description"`
	Trophies         int          `json:"trophies"`
	RequiredTrophies int          `json:"requiredTrophies"`
	Members          []ClubMember `json:"members"`
	Type             string       `json:"string"`
	BadgeID          int          `json:"badgeID"`
}

File: front/src/types/club.ts

export interface IClubMember {
    tag: string;
    name: string;
    trophies: number;
    role: string;
    nameColor: string;
}

export interface IClub {
    tag: string;
    name: string;
    description: string;
    trophies: number;
    requiredTrophies: number;
    members: Array<IClubMember>;
    type: string;
    badgeID: number;
}
